nncf/common/compression.py

```diff
@@ -10,18 +10,43 @@
  See the License for the specific language governing permissions and
  limitations under the License.
 """
+from abc import abstractmethod
+from typing import Any, Dict, List, Optional, Tuple
 
-from typing import Optional, List, Tuple, Any
-
-from nncf.api.compression import CompressionAlgorithmController
 import nncf.common.factory as factory
+from nncf.api.compression import CompressionAlgorithmBuilder
+from nncf.api.compression import CompressionAlgorithmController
+from nncf.api.compression import CompressionLevel
+from nncf.api.compression import ModelType
+from nncf.common.utils.logger import logger as nncf_logger
+from nncf.common.utils.registry import Registry
+
+
+class BaseControllerStateNames:
+    LOSS = 'loss_state'
+    SCHEDULER = 'scheduler_state'
+    COMPRESSION_STAGE = 'compression_stage'
+    COMPRESSION_LEVEL = 'compression_level'
 
 
 class BaseCompressionAlgorithmController(CompressionAlgorithmController):
     """
     Contains the implementation of the basic functionality of the compression controller.
     """
 
+    def __init__(self, target_model: ModelType):
+        """
+        Initializes the internal state of the compression algorithm controller.
+
+        :param target_model: The model with additional modifications necessary
+            to enable algorithm-specific compression during fine-tuning built
+            by the `CompressionAlgorithmBuilder`.
+        """
+        super().__init__(target_model)
```

Comment:
Could you please remove the `__init__()` method from the `CompressionAlgorithmController` class and use `self._model = target_model` here?

Reply:
NO! I won't fix it, because I didn't introduced this in `CompressionAlgorithmController`. 
It's design issue of NNCF API it should be fixed separately.

BTW, it's not that simple, because it most probably would require creation of base implementation of CompositeController to pass `model` in multi-inherent child classes.